iXBRLViewerPlugin/viewer/src/js/util.js

```diff
@@ -35,6 +35,13 @@ export function momentToHuman(d, adjust) {
     return d.format("D MMM Y HH:mm:ss");
 }
 
+/*
+ * Format a number with a thousands separator, infinite decimal places
+ */
+export function numberWithCommas(v) {
+    return Number(v).toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ",");
```

Comment:
Why is this regex different from `formatNumber`?  Perhaps just change `formatNumber` to this:
```
    var n = Number(v);
    var s = d === undefined ? n.toString() : n.toFixed(Math.max(0, d));
    return d.replace(...);
```
...and then remove all the special-case handling from `readableValue`?